linkerd/app/core/src/config.rs

```diff
@@ -1,1137 +1,70 @@
-    s.parse().map_err(|_| ParseError::NotANumber)
-}
-
-fn parse_duration(s: &str) -> Result<Duration, ParseError> {
-    use regex::Regex;
-
-    let re = Regex::new(r"^\s*(\d+)(ms|s|m|h|d)?\s*$").expect("duration regex");
-
-    let cap = re.captures(s).ok_or(ParseError::NotADuration)?;
-
-    let magnitude = parse_number(&cap[1])?;
-    match cap.get(2).map(|m| m.as_str()) {
-        None if magnitude == 0 => Ok(Duration::from_secs(0)),
-        Some("ms") => Ok(Duration::from_millis(magnitude)),
-        Some("s") => Ok(Duration::from_secs(magnitude)),
-        Some("m") => Ok(Duration::from_secs(magnitude * 60)),
-        Some("h") => Ok(Duration::from_secs(magnitude * 60 * 60)),
-        Some("d") => Ok(Duration::from_secs(magnitude * 60 * 60 * 24)),
-        _ => Err(ParseError::NotADuration),
-    }
-}
-
-fn parse_socket_addr(s: &str) -> Result<SocketAddr, ParseError> {
-    match parse_addr(s)? {
-        Addr::Socket(a) => Ok(a),
-        _ => {
-            error!("Expected IP:PORT; found: {}", s);
-            Err(ParseError::HostIsNotAnIpAddress)
-        }
-    }
-}
-
-fn parse_addr(s: &str) -> Result<Addr, ParseError> {
-    Addr::from_str(s).map_err(|e| {
-        error!("Not a valid address: {}", s);
-        ParseError::AddrError(e)
-    })
-}
-
-fn parse_port_set(s: &str) -> Result<IndexSet<u16>, ParseError> {
-    let mut set = IndexSet::new();
-    for num in s.split(',') {
-        set.insert(parse_number::<u16>(num)?);
-    }
-    Ok(set)
-}
-
-pub(super) fn parse_identity(s: &str) -> Result<identity::Name, ParseError> {
-    identity::Name::from_hostname(s.as_bytes()).map_err(|identity::InvalidName| {
-        error!("Not a valid identity name: {}", s);
-        ParseError::NameError
-    })
-}
-
-pub(super) fn parse<T, Parse>(
-    strings: &dyn Strings,
-    name: &str,
-    parse: Parse,
-) -> Result<Option<T>, Error>
-where
-    Parse: FnOnce(&str) -> Result<T, ParseError>,
-{
-    match strings.get(name)? {
-        Some(ref s) => {
-            let r = parse(s).map_err(|parse_error| {
-                error!("{}={:?} is not valid: {:?}", name, s, parse_error);
-                Error::InvalidEnvVar
-            })?;
-            Ok(Some(r))
-        }
-        None => Ok(None),
-    }
-}
-
-#[allow(dead_code)]
-fn parse_deprecated<T, Parse>(
-    strings: &dyn Strings,
-    name: &str,
-    deprecated_name: &str,
-    f: Parse,
-) -> Result<Option<T>, Error>
-where
-    Parse: Copy,
-    Parse: Fn(&str) -> Result<T, ParseError>,
-{
-    match parse(strings, name, f)? {
-        Some(v) => Ok(Some(v)),
-        None => {
-            let v = parse(strings, deprecated_name, f)?;
-            if v.is_some() {
-                warn!("{} has been deprecated; use {}", deprecated_name, name);
-            }
-            Ok(v)
-        }
-    }
+#[derive(Clone, Debug)]
+pub struct ProxyConfig<A: OrigDstAddr = SysOrigDstAddr> {
+    pub server: ServerConfig<A>,
+    pub connect: ConnectConfig,
+    pub router_capacity: usize,
```

Comment:
I wonder if it makes sense for this to be a `RouterConfig`? Not sure if it's really necessary.